### ISSUE TITLE
Highlighting breaks when there are parens or square brackets in rule

### ISSUE THREAD

**Issue by pokey** [mrob95/vscode-TalonScript#2]
Type: <b>Bug</b>

For example:

```talon
(whatever): "whatever"
```

The whole line goes gray

Extension version: 0.3.16
VS Code version: Code 1.76.1 (Universal) (5e805b79fcb6ba4c2d23712967df89a089da575b, 2023-03-08T16:32:09.831Z)
OS version: Darwin arm64 22.3.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M2 Max (12 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|2, 2, 3|
|Memory (System)|32.00GB (0.14GB free)|
|Process Argv|--crash-reporter-id 99d35f96-61a9-461b-a60d-f915d32f3eee|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vscaac:30438847
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30681850
azure-dev_surveyone:30548225
pyindex848:30662994
nodejswelcome1cf:30587006
3biah626:30602489
pyind779:30671433
f6dab269:30613381
pythonsymbol12:30671437
a9j8j154:30646983
vsccsb:30677849
vscodeenable:30660116
funwalk2:30682974
pythonms35cf:30680985

```

</details>

**Comment by mirober** [comment]
Ah, whoops. An enormous gap in my test coverage! Should be fixed in the latest version (0.3.17).

Thanks for the report

**Comment by pokey** [comment]
yep that looks better thanks!
